OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs

```diff
@@ -62,7 +65,7 @@ public ConnectionLogic(Widget widget, ConnectionTarget endpoint, Action onConnec
 			var panel = widget;
 			panel.Get<ButtonWidget>("ABORT_BUTTON").OnClick = () => { CloseWindow(); onAbort(); };
 
-			widget.Get<LabelWidget>("CONNECTING_DESC").GetText = () => $"Connecting to {endpoint}...";
+			widget.Get<LabelWidget>("CONNECTING_DESC").GetText = () => modData.Translation.GetFormattedMessage(ConnectingToEndpoint, Translation.Arguments("endpoint", endpoint));
```

Comment:
as i remember GetText will be called each render. 

this makes Widgets in general more heavy - heavy as they are. 

it is not needed here, but couldn't the translation be memoized? for in game widget this would be nice.

Reply:
> couldn't the translation be memoized?

It should, yes. The standard pattern for cases like this is:
```csharp
var connectingDesc = modData.Translation.GetFormattedMessage(ConnectingToEndpoint, Translation.Arguments("endpoint", endpoint));
widget.Get<LabelWidget>("CONNECTING_DESC").GetText = () => connectingDesc;
```

(we have `CachedTranform` for cases where the arguments may change dynamically, but this doesn't appear to be needed here).

Reply:
yes, but it would propose to move the cached transform into the translation module by defining a `TranslateString`. to avoid having cached transforms everywhere a ui string is used.